addon/models/node.js

```diff
@@ -139,5 +139,87 @@ export default OsfModel.extend(FileItemMixin, {
         );
         this.set('_dirtyRelationships.contributors.remove', []);
         return promise;
-    }
+    },
+
+    addChild(title, description, category) {
+        let child = this.store.createRecord('node', {
+            title: title,
+            category: category || 'project',
+            description: description || null
+        });
+
+        return this.store.adapterFor('node').ajax(this.get('links.relationships.children.links.related.href'), 'POST', {
+            data: child.serialize(),
+        }).then(resp => {
+            child.unloadRecord();
+            this.store.pushPayload(resp);
+            let created = this.store.peekRecord('node', resp.data.id);
+            this.get('children').pushObject(created);
+            return created;
+        });
+    },
+
+    addUnregisteredContributor(fullName, email, permission, isBibliographic, index=Number.MAX_SAFE_INTEGER) {
+        let user = this.store.createRecord('user', {
+            fullName: fullName,
+            username: email
+        });
+        // After user has been saved, add user as a contributor
+        return user.save().then(user => this.addContributor(user.id, permission, isBibliographic, index));
+
+    },
+
+    addContributor(userId, permission, isBibliographic, index=Number.MAX_SAFE_INTEGER) {
```

Comment:
When I call addContributor there are two items that get pushed into the store, one with the user id and one with the correct contributor id.
![screen shot 2016-08-17 at 9 58 22 am](https://cloud.githubusercontent.com/assets/9755598/17738953/5fbbe16e-6461-11e6-8f0c-f1245361bf24.png)